### ISSUE TITLE
Container Memory Usage Always 0 in 4.3.0

### ISSUE THREAD

**Issue by robd003** [docker/for-mac#6076]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 17C272C8-850F-4CEB-AE3C-9BC16229A0B0/20211203180843   -- Virtualization.framework
  - Diagnostics ID: 17C272C8-850F-4CEB-AE3C-9BC16229A0B0/20211204031030   -- Hypervisor.framework

### Expected behavior
Show memory usage of containers

### Actual behavior
Containers always show 0 bytes used

### Information
  - macOS Version: 12.1
  - Intel chip or Apple chip: M1 Max
  - Docker Desktop Version: 4.3.0

### Steps to reproduce the behavior
Start **any** container. All containers report 0 for memory usage. CPU, network and disk I/O are all reported correctly.

**Comment by robd003** [comment]
I've tried using Virtualization.framework and Virtualization.framework and both don't report memory usage.

Oddly if I use the cli docker stats container_name I can see the memory stats, but in the dashboard it's **always** **0**

**Comment by robd003** [comment]
Confirmed that this is broken on Intel Macs too.

It worked fine in 4.2.0, but broke with the 4.3.0 release.

**Comment by Avertae** [comment]
More input on this, in case containerized app depends on "/sys/fs/cgroup/memory/memory.limit_in_bytes" it fails to start with:
`cat: /sys/fs/cgroup/memory/memory.limit_in_bytes: No such file or directory`

4.2.0 was ok.

**Comment by stephen-turner** [comment]
Thanks for the report, we're looking into this.

**Comment by stephen-turner** [comment]
Also broken on Windows Hyper-V but not Windows WSL 2. We've added a release note and we're working on a fix.

**Comment by mat007** [comment]
We have now fixed this issue in Docker Desktop 4.4.2, thanks!

**Comment by robd003** [comment]
@mat007 Awesome! Upgrading right now to verify

**Comment by BrowduesMan85** [comment]
Hey there, I noticed this post today and upgraded to docker desktop 4.4.2, per the recommendation.

However, I still don't see both the `mem` and the `cpu` directories.

Note that I am on Mac, Catalina OS, using an Intel I9.

Here is what I see when I examine the contents of `/sys/fs/cgroup`:

```zsh
root@eaas-configrun-service-7cbc8744ff-7zvgf:/# cd /sys/fs/cgroup/
cgroup.controllers        cgroup.threads            cpuset.cpus.partition     hugetlb.1GB.rsvd.max      io.latency                memory.max                pids.current
cgroup.events             cgroup.type               cpuset.mems               hugetlb.2MB.current       io.max                    memory.min                pids.events
cgroup.freeze             cpu.max                   cpuset.mems.effective     hugetlb.2MB.events        io.stat                   memory.oom.group          pids.max
cgroup.max.depth          cpu.stat                  hugetlb.1GB.current       hugetlb.2MB.events.local  memory.current            memory.stat               rdma.current
cgroup.max.descendants    cpu.weight                hugetlb.1GB.events        hugetlb.2MB.max           memory.events             memory.swap.current       rdma.max
cgroup.procs              cpu.weight.nice           hugetlb.1GB.events.local  hugetlb.2MB.rsvd.current  memory.events.local       memory.swap.events        
cgroup.stat               cpuset.cpus               hugetlb.1GB.max           hugetlb.2MB.rsvd.max      memory.high               memory.swap.high          
cgroup.subtree_control    cpuset.cpus.effective     hugetlb.1GB.rsvd.current  io.bfq.weight             memory.low                memory.swap.max     
```     

Is this expected or is this still broken? In 4.2 (and prior versions) we still saw both the `cpu` and `mem` directories.

**Comment by Avertae** [comment]
Fix is not enabled OOB, which was blamed already [here](https://github.com/docker/for-mac/issues/6073#issuecomment-1012711538)
TLDR;
~/Library/Group Containers/group.com.docker/settings.json
`"deprecatedCgroupv1": true,`

**Comment by BrowduesMan85** [comment]
Thank you
